libbindgen/src/codegen/mod.rs

```diff
@@ -656,8 +656,30 @@ impl CodeGenerator for CompInfo {
         //
         // TODO: Generate layout tests for template specializations, yay!
         if self.has_non_type_template_params() ||
-           self.is_template_specialization() {
-            return;
+            self.is_template_specialization() {
+                let layout = item.kind().expect_type().layout(ctx);
+                let canonical_name = item.canonical_name(ctx);
+
+                if let Some(layout) = layout {
+                    let fn_name = format!("bindgen_test_layout_template_{}", canonical_name);
+                    let fn_name = ctx.rust_ident_raw(&fn_name);
+                    let ident = item.to_rust_ty(ctx);
+                    let prefix = ctx.trait_prefix();
+                    let size_of_expr = quote_expr!(ctx.ext_cx(),
+                                                   ::$prefix::mem::size_of::<$ident>());
+                    let align_of_expr = quote_expr!(ctx.ext_cx(),
+                                                    ::$prefix::mem::align_of::<$ident>());
+                    let size = layout.size;
+                    let align = layout.align;
+                    let item = quote_item!(ctx.ext_cx(),
```

Comment:
Probably we should try to reuse this code with the other tests, that way it's easier to change in the future if needed.